src/api/solr-query.js

```diff
@@ -51,14 +51,35 @@ const fieldToQueryFilter = (field) => {
   return null;
 };
 
-const buildQuery = (fields, mainQueryField) => fields
+const buildQuery = (fields, mainQueryField) => {
+  var query = fields
 // Do not include main query field in filter field query param.
   .filter((searchField) => (!Object.hasOwnProperty.call(searchField, "field") || (Object.hasOwnProperty.call(searchField, "field") && searchField.field !== mainQueryField)))
   .map(fieldToQueryFilter)
   .filter((queryFilter) => queryFilter !== null)
   .map((queryFilter) => `fq=${queryFilter}`)
   .join("&");
 
+  // find the value for all text search
+  var all_text_value = fields
+  .filter((searchField) => searchField.field === '*')
+  .map((all) => all.value);
+
+  // search all text type fields
+  var all_text = fields
+  .filter((searchField) => (!Object.hasOwnProperty.call(searchField, "field") || (Object.hasOwnProperty.call(searchField, "field") && searchField.field !== mainQueryField)))
+  .filter((searchField) => searchField.type === 'text' && searchField.field !== "*")
+  .map((Queryfields) => encodeURIComponent(all_text_value != "" ? `${Queryfields.field}:${all_text_value}` : ""))
+  .join(" OR ");
+
+  // enable all text field seach cross multiple text type
+  query = all_text_value == "" ? query : query == "" ? `fq=${all_text}` : query + "&" + `fq=${all_text}`;
```

Comment:
Check if all text fields search has value and add all text field names in front of value as a value of fq query parameter.